regolith/builders/coabuilder.py

```diff
@@ -3,40 +3,211 @@
+            if not pub.get("month"):
+                print("WARNING: {} is missing month".format(
+                    pub["_id"]))
+            if pub.get("month") == "tbd".casefold():
+                print("WARNING: month in {} is tbd".format(
+                    pub["_id"]))
+            yield pub
+
+
+def get_since_date(rc):
+    """Get the since_date from command."""
+    if isinstance(rc, str):
+        since_date = dt.datetime.strptime(rc, '%Y-%m-%d').date() - relativedelta(months=NUM_MONTHS)
+    else:
+        since_date = dt.date.today() - relativedelta(months=NUM_MONTHS)
+    return since_date
+
+
+def get_coauthos_from_pubs(pubs):
+    """Get co-authors' names from the publication."""
+    my_collabs = []
+    for pub in pubs:
+        my_collabs.extend(
+            [
+                collabs for collabs in
+                (names for names in pub.get('author', []))
+            ]
+        )
+    return my_collabs
+
+
+def query_people_and_instituions(rc, names):
+    """Get the people and institutions names."""
+    people, institutions = [], []
+    for person_name in names:
+        person_found = fuzzy_retrieval(all_docs_from_collection(
+            rc.client, "people"),
+            ["name", "aka", "_id"],
+            person_name)
+        if not person_found:
+            person_found = fuzzy_retrieval(all_docs_from_collection(
+                rc.client, "contacts"),
+                ["name", "aka", "_id"], person_name)
+            if not person_found:
+                print(
+                    "WARNING: {} not found in contacts. Check aka".format(
+                        person_name))
+            else:
+                people.append(person_found['name'])
+                inst = fuzzy_retrieval(all_docs_from_collection(
+                    rc.client, "institutions"),
+                    ["name", "aka", "_id"],
+                    person_found["institution"])
+                if inst:
+                    institutions.append(inst["name"])
+                else:
+                    institutions.append(person_found.get("institution", "missing"))
+                    print("WARNING: {} missing from institutions".format(
+                        person_found["institution"]))
+        else:
+            people.append(person_found['name'])
+            pinst = person_found.get("employment",
+                                     [{"organization": "missing"}])[
+                0]["organization"]
+            inst = fuzzy_retrieval(all_docs_from_collection(
+                rc.client, "institutions"), ["name", "aka", "_id"],
+                pinst)
+            if inst:
+                institutions.append(inst["name"])
+            else:
+                institutions.append(pinst)
+                print(
+                    "WARNING: {} missing from institutions".format(
+                        pinst))
+    return people, institutions
+
+
+def query_last_first_instutition_names(rc, ppl_names, excluded_inst_name=None):
```

Comment:
this is not really a query.  Maybe call it ``format_last_first_instutition_names`` ?